### ISSUE TITLE
Is it possible to get the number of items that passed checks, as well as the number of items that failed checks?

### ISSUE THREAD

**Issue by sehilyi** [dequelabs/axe-core#4283]
### Product

axe-core

### Question

I am trying to calculate the failure rate, i.e., `# of fails` / `# of potential fails`. For example, for alt texts, I want to see how many images do not have alt texts (`# of fails`) among all images (`# of potential fails`). Is it possible to get such information through axe-core, i.e., the number of items checked per rule?

**Comment by straker** [comment]
Thanks for the issue. Sorry for the late reply, I've been on vacation. You can probably get at this information by looking at all there returned results (`passes`, `incomplete`, and `violations`) for a particular rule. So for example the `image-alt` rule you can get all nodes the rule ran against by summing the number of nodes in each section, then looking at the `violations` array you can get the number of failed nodes.

```js
const results = await axe.run()
const totalNodes = ['passes', 'incomplete', 'violations'].reduce((count, type) => {
  const rule = results[type].find(rule => rule.id === 'image-alt');
  return count + (rule ? rule.nodes.length : 0)
}, 0)
const failedRule = results.violations.find(rule => rule.id === 'image-alt');
const failedNodes = failedRule ? failedRule.nodes.length : 0;

const failureRate = failedNodes / totalNodes;
```
